core/http/middleware/auth.go

```diff
@@ -0,0 +1,94 @@
+package middleware
+
+import (
+	"crypto/subtle"
+	"errors"
+	"slices"
+
+	"github.com/dave-gray101/v2keyauth"

```

Comment:
Since github.com/dave-gray101/v2keyauth is really about a single file, I'd suggest to move it under `pkg/` rather then having another dependency

Reply:
I pulled it out because I have other personal projects that will be using it - if you'd like me to vendor it here as well, I can - it's not like I'm expecting to make significant changes... just lazy to maintain in multiple places if the upstream fiber team makes any changes I need to port backwards.

Reply:
Also - I just left an exact copy of the Fiber license / copyright in that Repo - since a lot of the code originated within fiber, rather than deal with split copyrights within the repo I figured it would be easier to leave it "loose" so that file would get copied to the correct folder...

I am **not** a lawyer - just trying to play things safe.